Sources/Jobs/JobQueueHandler.swift

```diff
@@ -112,15 +121,33 @@ final class JobQueueHandler<Queue: JobQueueDriver>: Sendable {
                     self.updateJobMetrics(for: job.name, startTime: startTime, error: error)
                     return
                 } catch {
-                    if count <= 0 {
+                    if job.remainingAttempts <= 0 {
                         logger.debug("Job failed")
                         try await self.queue.failed(jobId: queuedJob.id, error: error)
                         self.updateJobMetrics(for: job.name, startTime: startTime, error: error)
                         return
                     }
-                    count -= 1
+                    // count -= 1
                     logger.debug("Retrying Job")
                     self.updateJobMetrics(for: job.name, startTime: startTime, retrying: true)
+                    let attempts = job.attempts + 1
+                    let delay = self.backoff(attempts: attempts)
+                    // let nextJobQueueDuration = Date.now.timeIntervalSince(job.queuedAt) + delay
+                    // once this is parameters are figured out. extact into a function so it can be used for both
+                    // delayed and retries
+                    let jobBuffer = EncodableJob(
+                        id: JobIdentifier(job.name),
+                        // TODO: How do I get job parameters from here?
+                        parameters: queuedJob,
```

Comment:
@adam-fowler or @Joannis I need some help here  please when you have a chance. How can I get access to job parameters here? 

I have want to move this operation into a function so that it can be used [here too](https://github.com/hummingbird-project/swift-jobs/pull/21/files#diff-ca3bd900c8cd6d93c3f9a8a1794fc50dcd394edd6351f2bf9acfb42d564ec3e7R106)

Reply:
Add to `JobInstanceProtocol`
```swift
    /// Job parameters
    var parameters: Parameters { get }
```
and add to `JobInstance`
```swift
    /// job parameters
    var parameters: Parameters { self.data.parameters }
```
then they are accessible as `job.parameters`